packages/socket/lib/browser.ts

```diff
@@ -4,18 +4,26 @@ import type { SocketShape } from './types'
 
 export type { Socket } from 'socket.io-client'
 
-const sockets: {[key: string]: CDPBrowserSocket} = {}
+declare global {
+  interface Window {
+    cypressSockets: {[key: string]: CDPBrowserSocket}
+  }
+}
+
+window.cypressSockets ||= {}
 let chromium = false
 
 export function client (uri: string, opts?: Partial<ManagerOptions & SocketOptions>): SocketShape {
   if (chromium) {
     const fullNamespace = `${opts?.path}${uri}`
 
-    if (!sockets[fullNamespace]) {
-      sockets[fullNamespace] = new CDPBrowserSocket(fullNamespace)
+    if (!window.cypressSockets[fullNamespace]) {
+      window.cypressSockets[fullNamespace] = new CDPBrowserSocket(fullNamespace)
     }
 
-    return sockets[fullNamespace]
+    window.cypressSockets[fullNamespace].connect()
```

Comment:
What does connect do that we weren't doing before and why does it fix the hang?

Reply:
Added some doc to explain this a little better: [`27b44ed` (#27781)](https://github.com/cypress-io/cypress/pull/27781/commits/27b44ed1a67b17e529b5e8016c8470587cc9f239)